components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSimplePAPStatusDataHandler.java

```diff
@@ -0,0 +1,389 @@
+            if (!useLastStatusOnly) {
+                Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+                try {
+                    // Get the existing status count
+                    int statusCount = getStatusCount(connection, about, key);
+
+                    // Delete old status data if the count exceeds the maximum records
+                    if (statusCount > maxRecords) {
+                        deleteStatus(connection, about, key, statusCount);
+                    }
+                    IdentityDatabaseUtil.commitTransaction(connection);
+                } catch (SQLException e) {
+                    IdentityDatabaseUtil.rollbackTransaction(connection);
+                    throw new EntitlementException("Error while deleting surplus policy status", e);
+                } finally {
+                    IdentityDatabaseUtil.closeConnection(connection);
+                }
+            }
+        }
+    }
+
+    private List<StatusHolder> getStatus(String key, String about) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        List<StatusHolder> statusHolders = new ArrayList<>();
+        String query = EntitlementConstants.Status.ABOUT_POLICY.equals(about)
+                ? GET_POLICY_STATUS_SQL
+                : GET_SUBSCRIBER_STATUS_SQL;
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (NamedPreparedStatement getStatusPrepStmt = new NamedPreparedStatement(connection, query)) {
+                getStatusPrepStmt.setString(KEY, key);
+                getStatusPrepStmt.setInt(TENANT_ID, tenantId);
+                try (ResultSet statusSet = getStatusPrepStmt.executeQuery()) {
+                    while (statusSet.next()) {
+                        StatusHolder statusHolder = new StatusHolder(about);
+                        if (EntitlementConstants.Status.ABOUT_POLICY.equals(about)) {
+                            statusHolder.setKey(statusSet.getString(POLICY_ID));
+                        } else {
+                            statusHolder.setKey(
+                                    statusSet.getString(SUBSCRIBER_ID));
+                        }
+                        statusHolder.setType(statusSet.getString(STATUS_TYPE));
+                        statusHolder.setSuccess(statusSet.getBoolean(IS_SUCCESS));
+                        statusHolder.setUser(statusSet.getString(USER));
+                        statusHolder.setTarget(statusSet.getString(TARGET));
+                        statusHolder.setTargetAction(statusSet.getString(TARGET_ACTION));
+                        statusHolder.setTimeInstance(String.valueOf(statusSet.getTimestamp(LOGGED_AT).getTime()));
+                        statusHolder.setMessage(statusSet.getString(MESSAGE));
+
+                        String version = null;
+                        if (EntitlementConstants.Status.ABOUT_POLICY.equals(about) &&
```

Comment:
Lets reuse the above if condition [1] so that status type related logic not scattered.

[1] https://github.com/wso2/carbon-identity-framework/pull/5826/files#diff-56d5efe3772d833b1477e98e1a7641115183bff2c2ac146d5a2bad6a171dad4bR220